dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/ForkingClusterInvoker.java

```diff
@@ -96,7 +93,7 @@ public Result doInvoke(final Invocation invocation, List<Invoker<T>> invokers, L
                 });
             }
             try {
-                Object ret = ref.poll(timeout, TimeUnit.MILLISECONDS);
+                Object ret = ref.poll(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
```

Comment:
I don't think it's reasonable to make the queue wait indefinitely, which may generate some unknown exceptions, and a reasonable way would be to reverse the relevant exception message when the ret is empty

Reply:
Thank you for your suggestion. I understand what you mean, but if the provider side times out, an exception will be caught here and put into the queue, so the poll method will not return any null. There should be no other exceptions。

`                    try {
                        Result result = invokeWithContextAsync(invoker, invocation, consumerUrl);
                        ref.offer(result);
                    } catch (Throwable e) {
                        int value = count.incrementAndGet();
                        if (value >= selected.size()) {
                            ref.offer(e);
                        }
                    }`

Reply:
I agree with what you said